src/main/java/seedu/address/commons/util/StringUtil.java

```diff
@@ -65,4 +65,20 @@ public static boolean isNonZeroUnsignedInteger(String s) {
             return false;
         }
     }
+
+    /**
+     * Filters nonempty strings from the given strings and concatenates them with newline.
+     * @param strings Strings to be joined.
+     * @return Concatenated string.
+     */
+    public static String join(String ...strings) {
+        StringBuilder builder = new StringBuilder();
+        for (String s : strings) {
+            if (s.length() > 0) {
+                builder.append(s.trim());
+                builder.append("\n");
+            }
+        }
+        return builder.toString();
```

Comment:
Could we use `String.join` instead?

Reply:
This was added to make concatenate strings in a more standardized manner. The method is meant to be different from the regular String.join(), in the sense that it allows you to join two strings possibly ending with newline, and ensuring that there is only 1 newline character between them.
This may make testing easier, so that tests will not fail due to an extra (or missing) newline.
However, whether this will see regular usage remains to be seen.

Reply:
Ah you are right, I see what you mean

Reply:
Thinking about it more, this feels to me like a slightly unprecise way of joining strings. And the naming may make others confuse it with the String.join function. If this is just to make tests pass, could we make our tests more precise instead?

This might be a bit nitpicky, so I'll leave it as an optional suggestion